### PR TITLE
Fixed removing items from the shell causing a crash

### LINKED ISSUE #9938: Shell crashes upon adding/removing items
### Description

Upon adding or removing FlyoutItems during runtime, the Shell occasionally crashes with nullReferenceExceptions

#### 1. Add Items:
- Start App
- click on `disable add workaround`
- click on `add Shell Item`
- click on `add Shell Item`
=> App will crash

#### 2. Remove Items:
- Start App
- click on `disable remove workaround`
- click on `add Shell Item`
- click on `add Shell Item`
- click on `add Shell Item`
- click on `remove Shell Item`
- click on `remove Shell Item`
- click on `remove Shell Item`
=> App will crash

### Link to public reproduction project repository

https://github.com/rxDietel/MauiAppShellTest

### Version with bug

6.0.486 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows, iOS, I was *not* able test on other platforms

### Affected platform versions

net6.0-windows10.0.19041.0
iOS 15.6.1

### Did you find any workaround?

See Bug number 4 in ReadMe.md of reproduction project

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Foda**
### Description of Change

This PR fixes a few null-refs when removing items from the shell, either when the item is selected or not.

Note: fixing this issue presents another problem. Removing shell items that are currently selected will not set the selection to a non-removed item. It's therefore recommended that users set the selected shell item to something else before removing the item.

Note2: I've only tested this change on Windows. 

### Issues Fixed

Fixes #9938

### CODE REVIEW

**Review by PureWeen — CHANGES_REQUESTED** [review]
Once this is [merged](https://github.com/dotnet/maui/pull/15220) can you rebase this PR?

File: src/Controls/src/Core/Shell/ShellItem.cs [hunk 1]
```diff
@@ -109,10 +109,14 @@ bool IShellItemController.ShowTabs
 				// Windows supports nested tabs so we want the tabs to display
 				// if the current shell section has multiple contents
 				if (ShellItemController.GetItems().Count > 1 ||
-					(CurrentItem as IShellSectionController).GetItems().Count > 1)
+					(CurrentItem != null && (CurrentItem as IShellSectionController).GetItems().Count > 1))
```
Comment (PureWeen) [inline src/Controls/src/Core/Shell/ShellItem.cs:112]: I think we can just add some null terminator to make this work

`(CurrentItem as IShellSectionController)?.GetItems()?.Count > 1)`

File: src/Controls/tests/Core.UnitTests/ShellTests.cs [hunk 2]
```diff
@@ -255,6 +256,36 @@ public void CurrentItemDoesNotChangeOnSecondAdd()
 			Assert.Equal(shellItem, shell.CurrentItem);
 		}
 
+		[Fact]
+		public void AddRemoveItemsDoesNotCrash()
```
Comment (PureWeen) [inline src/Controls/tests/Core.UnitTests/ShellTests.cs:260]: If I'm following the code in this PR correctly, I don't think this UnitTest adds any coverage to the added code. 

Both code chunks added will only execute when they are running as a WinUI app. 

Can we add the repro [here](https://github.com/dotnet/maui/issues/9938) to a device test and then work from there?

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/tests/Core.UnitTests/ShellTests.cs (hunk 2 above)
Comment (Foda) [reply src/Controls/tests/Core.UnitTests/ShellTests.cs:260]: Just got back from vacation -- for sure will add a device test.

File: src/Controls/src/Core/Handlers/Shell/Windows/ShellView.cs [hunk 3]
```diff
@@ -209,10 +209,13 @@ internal void SwitchShellItem(ShellItem newItem, bool animate = true)
 			{
 				if (navItems.TryGetWithData(newItem, out NavigationViewItemViewModel vm1))
 					SelectedItem = vm1;
-				else if (navItems.TryGetWithData(newItem.CurrentItem, out NavigationViewItemViewModel vm2))
-					SelectedItem = vm2;
-				else if (navItems.TryGetWithData(newItem.CurrentItem.CurrentItem, out NavigationViewItemViewModel vm3))
-					SelectedItem = vm3;
+				else if (newItem.CurrentItem != null)
```
Comment (PureWeen) [inline src/Controls/src/Core/Handlers/Shell/Windows/ShellView.cs:212]: `is not null`

File: src/Controls/tests/DeviceTests/Elements/Shell/ShellTests.Windows.cs [hunk 4]
```diff
@@ -519,6 +519,64 @@ public async Task ShellToolbarWithOnlyToolbarItemsIsVisible()
...
+			await CreateHandlerAndAddToWindow<ShellHandler>(shell, (handler) =>
+			{
+				rootItem.IsVisible = true;
+
+				shell.CurrentItem = rootItem.Items[itemCount - 1];
+
+				// Remove all root child items
+				for (int i = 0; i < itemCount; i++)
+				{
+					shell.Items.ElementAt(1).Items.RemoveAt(0);
```
Comment (PureWeen) [inline src/Controls/tests/DeviceTests/Elements/Shell/ShellTests.Windows.cs:570]: Can we reuse some of the named items above? 

`ElementAt(1)` is just `rootItem` yea?

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/tests/DeviceTests/Elements/Shell/ShellTests.Windows.cs [hunk 5]
```diff
@@ -519,6 +519,64 @@ public async Task ShellToolbarWithOnlyToolbarItemsIsVisible()
 			});
 		}
 
+		[Fact(DisplayName = "Shell Add then Remove Items")]
+		public async Task ShellAddRemoveItems()
```
Comment (PureWeen) [inline src/Controls/tests/DeviceTests/Elements/Shell/ShellTests.Windows.cs:523]: Can we move this to `ShellTests` so it runs on all platforms? Nothing here is specific to `Windows` so I think it'll add some coverage on `iOS` and `Android` as well. 

I tested on `Android` and it passes. I feel good about iOS :-)

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/tests/DeviceTests/Elements/Shell/ShellTests.Windows.cs (hunk 5 above)
Comment (Foda) [reply src/Controls/tests/DeviceTests/Elements/Shell/ShellTests.Windows.cs]: Moved this + addressed above feedback :)

**Review by PureWeen — APPROVED** [review]

**Pull request merged**